src/Php80/Php80.php

```diff
@@ -57,6 +57,15 @@ public static function get_debug_type($value): string
         return (get_parent_class($class) ?: key(class_implements($class)) ?: 'class').'@anonymous';
     }
 
+    public static function get_resource_id($res): int
+    {
+        if (@\get_resource_type($res) === null) {
```

Comment:
let's optimize for the happy path:
`if (!\is_resource($res) && null === @get_resource_type($res)) {`

Reply:
do we even need the `get_resource_type` check ?

Reply:
we need it for closed resources yes, that's an odd edge case.